staging/src/k8s.io/kubectl/pkg/validation/schema.go

```diff
@@ -101,3 +103,32 @@ func (c ConjunctiveSchema) ValidateBytes(data []byte) error {
 	}
 	return utilerrors.NewAggregate(list)
 }
+
+// ParamVerifyingSchema only performs validation
+// based on the fieldValidation query param
+// being unsupported by the apiserver, because
+// server-side validation will be performed instead
+// of client-side validation.
+type ParamVerifyingSchema struct {
+	Schema   Schema
+	Verifier *resource.QueryParamVerifier
+}
+
+// ValidateBytes validates bytes per a ParamVerifyingSchema
+func (c ParamVerifyingSchema) ValidateBytes(data []byte) error {
```

Comment:
Yea, I've added a unit test to specifically test the "HasSupport" logic. It doesn't seem appropriate to test the schemavalidation functions here because that is already captured by testing in the `schemavalidation` package.

As you can see, this test has also forced me to make the QueryParamVerifier an interface that we can mock out. I took the approach of only using the interface for where we explicitly need it for testing and left the rest of the uses in code to use the concrete `QueryParamVerifier` type still. Let me know if that's okay.